services/app-api/handlers/formTemplates/write.ts

```diff
@@ -15,24 +17,46 @@ export const writeFormTemplate = handler(async (event, _context) => {
       body: UNAUTHORIZED_MESSAGE,
     };
   }
-  const body = JSON.parse(event!.body!);
-  if (!body?.formTemplateId) {
+  const unvalidatedPayload = JSON.parse(event!.body!);
+  if (!unvalidatedPayload?.formTemplateId) {
     throw new Error(NO_KEY_ERROR_MESSAGE);
-  } else if (!body?.formTemplate) {
+  } else if (!unvalidatedPayload?.formTemplate) {
     throw new Error(MISSING_DATA_ERROR_MESSAGE);
   } else {
-    const params = {
-      TableName: process.env.FORM_TEMPLATE_TABLE_NAME!,
-      Item: {
-        formTemplateId: body.formTemplateId,
-        createdAt: Date.now(),
-        lastAltered: Date.now(),
-        lastAlteredBy: event?.headers["cognito-identity-id"],
-        formTemplate: body.formTemplate,
-        ...body,
-      },
-    };
-    await dynamoDb.put(params);
-    return { status: StatusCodes.SUCCESS, body: params };
+    const validationSchema = yup.object().shape({
+      formTemplateId: yup.string().required(),
+      formTemplateVersion: yup.string().required(),
+      formTemplate: yup
+        .object()
+        .required()
+        .shape({
+          name: yup.string().required(),
+          basePath: yup.string().required(),
+          version: yup.string().required(),
+          routes: yup.array().of(yup.object()).required(),
+          validationSchema: yup.object().required(),
+        }),
+    });
+
+    const validatedPayload = await validateData(
+      validationSchema,
+      unvalidatedPayload
+    );
+
+    if (validatedPayload) {
+      const params = {
+        TableName: process.env.FORM_TEMPLATE_TABLE_NAME!,
+        Item: {
+          formTemplateId: unvalidatedPayload.formTemplateId,
+          createdAt: Date.now(),
+          lastAltered: Date.now(),
+          lastAlteredBy: event?.headers["cognito-identity-id"],
+          formTemplate: unvalidatedPayload.formTemplate,
```

Comment:
same note as above: using validatedPayload to read instead

Reply:
👍